File: read_resolutions.py
import os
import struct

def get_mp4_resolution(filepath):
    # Search for 'tkhd' atom in MP4
    with open(filepath, 'rb') as f:
        data = f.read()
        
    tkhd_idx = data.find(b'tkhd')
    if tkhd_idx != -1:
        # tkhd version & flags is 4 bytes
        # creation time, modification time, track ID, reserved is 16 bytes (or 32 bytes if version 1)
        # We need to check the version first.
        # tkhd_idx is at the start of 'tkhd' type. The size of atom is 4 bytes before it.
        # Let's parse version which is at tkhd_idx + 4
        version = data[tkhd_idx + 4]
        if version == 1:
            # 64-bit offsets
            # version & flags: 4
            # creation time: 8
            # modification time: 8
            # track id: 4
            # reserved: 4
            # duration: 8
            # reserved: 8
            # layer & alternate group: 4
            # volume & reserved: 4
            # matrix: 36
            # width: 4 (fixed point 16.16)
            # height: 4 (fixed point 16.16)
            offset = tkhd_idx + 4 + 4 + 8 + 8 + 4 + 4 + 8 + 8 + 4 + 4 + 36
        else:
            # 32-bit offsets
            # version & flags: 4
            # creation time: 4
            # modification time: 4
            # track id: 4
            # reserved: 4
            # duration: 4
            # reserved: 8
            # layer & alternate group: 4
            # volume & reserved: 4
            # matrix: 36
            # width: 4
            # height: 4
            offset = tkhd_idx + 4 + 4 + 4 + 4 + 4 + 4 + 8 + 4 + 4 + 36
            
        if offset + 8 <= len(data):
            width_fp = struct.unpack('>I', data[offset:offset+4])[0]
            height_fp = struct.unpack('>I', data[offset+4:offset+8])[0]
            return width_fp >> 16, height_fp >> 16
    return None

workspace_dir = r"c:\Users\wnals\OneDrive\바탕 화면\new\beauty-booth"
female_path = os.path.join(workspace_dir, "assets", "avatar_female.mp4")
male_path = os.path.join(workspace_dir, "assets", "avatar_male.mp4")

print("Female video resolution:", get_mp4_resolution(female_path))
print("Male video resolution:", get_mp4_resolution(male_path))

File: read_resolutions.js
const fs = require('fs');
const path = require('path');

function getMp4Resolution(filepath) {
  try {
    const data = fs.readFileSync(filepath);
    const tkhdIdx = data.indexOf(Buffer.from('tkhd'));
    if (tkhdIdx !== -1) {
      // tkhd atom starts at tkhdIdx - 4 (if size is 4 bytes)
      // but let's just use tkhdIdx which points to 'tkhd' (4 bytes)
      const version = data[tkhdIdx + 4];
      let offset;
      if (version === 1) {
        // version & flags (4) + creation (8) + modification (8) + track_id (4) + reserved (4) + duration (8) + reserved (8) + layer/alt (4) + volume/res (4) + matrix (36)
        offset = tkhdIdx + 4 + 4 + 8 + 8 + 4 + 4 + 8 + 8 + 4 + 4 + 36;
      } else {
        // version & flags (4) + creation (4) + modification (4) + track_id (4) + reserved (4) + duration (4) + reserved (8) + layer/alt (4) + volume/res (4) + matrix (36)
        offset = tkhdIdx + 4 + 4 + 4 + 4 + 4 + 4 + 8 + 4 + 4 + 36;
      }
      if (offset + 8 <= data.length) {
        const widthVal = data.readUInt32BE(offset);
        const heightVal = data.readUInt32BE(offset + 4);
        const width = widthVal >> 16;
        const height = heightVal >> 16;
        const widthFraction = widthVal & 0xFFFF;
        const heightFraction = heightVal & 0xFFFF;
        return { width: width + widthFraction / 65536, height: height + heightFraction / 65536, version };
      }
    }
  } catch (err) {
    console.error(err);
  }
  return null;
}

const workspaceDir = "c:\\Users\\wnals\\OneDrive\\바탕 화면\\new\\beauty-booth";
const femalePath = path.join(workspaceDir, "assets", "avatar_female.mp4");
const malePath = path.join(workspaceDir, "assets", "avatar_male.mp4");

console.log("Female video resolution:", getMp4Resolution(femalePath));
console.log("Male video resolution:", getMp4Resolution(malePath));
